tests/quick.rs

```diff
@@ -258,12 +258,12 @@ where
         let mut it = get_it();
 
         for _ in 0..(counts.len() - 1) {
-            if let None = it.next() {
+            if it.next().is_none() {
                 panic!("Iterator shouldn't be finished, may not be deterministic");
             }
```

Comment:
When we're at it: Shouldn't this be an `assert!(it.next().is_none(), "Iterator shouldn't be finished, may not be deterministic")`?

Reply:
`is_some` but yeah. Strange that clippy didn't lint this. I will investigate it :)